### ISSUE TITLE
Release .exe executable build

### ISSUE THREAD

**Issue by zimonitrome** [sherlock-project/sherlock#2006]
## Checklist

- [x] I'm reporting a feature request
- [x] I've checked for similar feature requests including closed ones

## Description

It has come to my attention that some people attemptng to use this repo are foreign to using Python.

Some people don't care too much about the code. They just want to download this amazing application and use it.

They might not understand why there's code, but would appreciate if a .EXE file was made and was simply handed to them. The dedicated maintainers might expect everyone to be developers that understands code. Well, some are note and they don't understand it. They only know to download and install applications. So what's the solution? Make an EXE file and give it to them. Again, thanks to the dedicated maintainers!

**Comment by OrdinaryEnder** [comment]
Just install python and do the 4 steps it's not that hard bro

**Comment by master117** [comment]
> Just install python and do the 4 steps it's not that hard bro

It absolutely is, you just don't understand how un-tech-savy the average person is. Experience Hell

**Comment by ohaiibuzzle** [comment]
My two cents (I've been observing the situation on Reddit): Probably not worth it.

Providing binaries would instantly lower the barrier to entry and increase the amount of people who don't know what they are doing, attempting to use Sherlock, not understanding errors messages and create issues for otherwise trivial questions.

And
> how un-tech-savy the average person

I don't see why "the average person" would suddenly carry out an OSINT operation?

**Comment by garandal245** [comment]
if a user cannot handle installing python and the steps they are 100% going to be unable to properly use a command line program properly and at that point to satisfy that you would need to actually make a UI which i imagine is something that the sherlock maintainers dont find valuable for an OSINT tool not even meant for a non tech savy person

**Comment by decadent** [comment]
The Python nature of this project inadvertently works as a minimal filter against script kiddie wannabes that encountered it somewhere as 'a tool to stalk someone online'. Not only it's advisable to raise the barrier of entry even further, but what you propose is directly immoral and despicable, and you should be banned from here with all speediness.

**Comment by zimonitrome** [comment]
I originally made this issue as a tongue in cheek joke and nod to the Reddit post, but I'm gonna try to argue for it nevertheless.

"un-tech-savy people" are already trying to use this repository, evident by some of the hilarious issues. Why don't we think that lowering the barrier for using the software would lead to these users having a successful experience (as in fulfilling their goals) and therefore not make these silly issues?

> I don't see why "the average person" would suddenly carry out an OSINT operation? (Unless they are trying to find someone's social accounts and come to Sherlock from the Google keyword "find usernames across social networks", in which case, please get the binaries as far away from those as possible)

The first and only sentence describing this repo is literally "Hunt down social media accounts by username across social networks". Claiming that this repo is some "extremely complex OSINT resource" is a bit silly, since all of this can pretty much be done through Google.

Nevertheless it *is* a neat tool that speeds up the process.

I don't see why authors of a stable and popular open source project *wouldn't* want more people to have access to their tool.

As far as user interface is concerned, the binary app could simply open the default shell, prompt the user for a username, load and display social media accounts, then wait for the user to enter a specific key to exit. No grapical UI needed, and no other options since I doubt many of these users would want to use them anyway.

**Comment by decadent** [comment]
> Claiming that this repo is some "extremely complex OSINT resource" is a bit silly, since all of this can pretty much be done through Google.

It's already evident from the word-salad issues that these people have the hacking chops of an ape with a purpose to stalk someone. Ascribing more intelligence to them than they have just makes the matter seem lighter than it is.

If you ever saw the deluge of creepy messages from randoms that women get on dating resources and social networks, you'd know how serious it is. But you don't.

**Comment by DontEatOreo** [comment]
I don't think this tool was ever meant for an average person, as it's strictly a CLI tool. Even if an `.EXE` file was made available, it would still not be useful to people who can't even install it.

Also, a tool like this SHOULD have some barrier to entry, and most people who are requesting `.EXE` probably aren't using it for anything good.

**Comment by tornikeo** [comment]
Here's an idea. We should add a joke .exe file that immediately opens a rickroll in the browser. That'd be fun.

**Comment by bjd89** [comment]
If you dont understand the terminal maybe this isnt for you?

**Comment by Eugeny** [comment]
Because this repo doesn't need an influx of morons asking how to hAcK An iNsTaGraM

**Comment by Rosuav** [comment]
> We should add a joke .exe file that immediately opens a rickroll in the browser. That'd be fun.

Or even better: An EXE file that deliberately listens on a random socket (to trigger personal firewall warnings), then pops up something telling them "Spyware successfully installed" or something. Not that it ACTUALLY does anything harmful, of course, but it says so, as a reminder of what happens when you download binaries that you don't understand.

At least with a .py file, you can skim over the code before running it.

**Comment by Tovape** [comment]
> > We should add a joke .exe file that immediately opens a rickroll in the browser. That'd be fun.
> 
> Or even better: An EXE file that deliberately listens on a random socket (to trigger personal firewall warnings), then pops up something telling them "Spyware successfully installed" or something. Not that it ACTUALLY does anything harmful, of course, but it says so, as a reminder of what happens when you download binaries that you don't understand.
> 
> At least with a .py file, you can skim over the code before running it.

At this point just upload bonzi buddy renamed as sherlock.exe

**Comment by Chizaruu** [comment]
Let them cook.

https://github.com/Chizaruu/sherlock/releases/tag/v69

**Comment by iderpHoi** [comment]
> Let them cook.
> 
> https://github.com/Chizaruu/sherlock/releases/tag/v69

mannnnn just talk in first person

**Comment by zimonitrome** [comment]
> Let them cook.
>
> https://github.com/Chizaruu/sherlock/releases/tag/v69

Amazing. Got that py2exe 400+MB binary 😍

**Comment by iustusae** [comment]
At this point just rewrite the whole shebang in c++ and watch the normies mald to their demise.

we will be there.

**Comment by yumio7** [comment]
Since the python is so hard to get running you might as well just write the whole thing in asm.

**Comment by Elyahu41** [comment]
No .msi installer?

**Comment by MCPO-Spartan-117** [comment]
Windows users are amusing sometimes.

**Comment by adyanth** [comment]
Halfway through the download for lolz, I remembered I daily a MacBook :)

**Comment by aidzmtsk** [comment]
The worst part of this issue is that I can't tell if he's trolling :laughing:

**Comment by FilipAndersson245** [comment]
> At this point just rewrite the whole shebang in c++ and watch the normies mald to their demise.
> 
> we will be there.

Let's rewrite it in rust 🦀🦀🦀

**Comment by FairyTail2000** [comment]
> > At this point just rewrite the whole shebang in c++ and watch the normies mald to their demise.
> > we will be there.
> 
> Let's rewrite it in rust 🦀🦀🦀

I second this

**Comment by Ryuo1** [comment]
What if we make it in C++ and force users to install 10 GB of build tools just to compile this project into a .exe file

**Comment by IverCoder** [comment]
This effort is not enough. We should upload this to the following storefronts for all of you lazy guys and gals out there:

- [ ] [Microsoft Store](https://learn.microsoft.com/en-us/windows/apps/publish/publish-your-app/overview?pivots=store-installer-msix) for Windows
- [ ] [Flathub](https://docs.flathub.org/docs/category/for-app-authors) for Linux
- [ ] App Store [for macOS](https://developer.apple.com/macos/submit/)
- [ ] App Store for [iOS](https://developer.apple.com/ios/submit/) and [iPadOS](https://developer.apple.com/ipados/submit/)
- [ ] [Self-hosted AltStore source](https://faq.altstore.io/distribute-your-apps/make-a-source) for iOS
- [ ] [Google Play Store](https://support.google.com/googleplay/android-developer/answer/9859751?hl=fil) for Android
- [ ] [F-Droid](https://f-droid.org/docs/Submitting_to_F-Droid_Quick_Start_Guide/) and [IzzyOnDroid](https://gitlab.com/IzzyOnDroid/repo/) for Android

**Comment by heipl-paul** [comment]
.dmg when?

**Comment by Ryuo1** [comment]
A gentoo variant needed too!

**Comment by IverCoder** [comment]
> > This effort is not enough. We should upload this to the following official and well-known storefronts for maximum user-friendliness:
> > 
> > * [ ]   [Microsoft Store](https://learn.microsoft.com/en-us/windows/apps/publish/publish-your-app/overview?pivots=store-installer-msix) for Windows
> > * [ ]   [Flathub](https://docs.flathub.org/docs/category/for-app-authors) for Linux
> > * [ ]   App Store [for macOS](https://developer.apple.com/macos/submit/)
> > * [ ]   App Store for [iOS](https://developer.apple.com/ios/submit/) and [iPadOS](https://developer.apple.com/ipados/submit/)
> > * [ ]   [Self-hosted AltStore source](https://faq.altstore.io/distribute-your-apps/make-a-source) for iOS
> > * [ ]   [Google Play Store](https://support.google.com/googleplay/android-developer/answer/9859751?hl=fil) for Android
> > * [ ]   [F-Droid](https://f-droid.org/docs/Submitting_to_F-Droid_Quick_Start_Guide/) and [IzzyOnDroid](https://gitlab.com/IzzyOnDroid/repo/) for Android
> 
> This is not enough for Linux. It should provide:
> 
> * [ ]  AppImage
> * [ ]  .deb
> * [ ]  .tar.gz
> * [ ]  .rpm
> * [ ]  Snap
> * [ ]  Nix
> * [ ]  bash install script
> * [ ]  fish install script

I'd rather they spent the resources in publishing to the FreeBSD Ports instead. We can't let Linux become the new Windows (in terms of software availability), can we?

**Comment by userbradley** [comment]
Can we also include a package for my [Samsung smart fridge](https://www.samsung.com/uk/refrigerators/side-by-side/rs8000nc-side-by-side-refrigerator-with-family-hub-591l-silver-rs6ha8880s9-eu/) whilst we're at it? It shouldn't be too hard, right? My friend who does computers said it should be easy and only take 10 minutes

**Comment by Ruffy314** [comment]
Maybe an .exe could require the user to create / login to an account on each of the scanned platforms.  
Manually, each time  
And ask the user if they want to inform the scanned person via DM that they have been looked up

**Comment by polidario** [comment]
Can someone burn it or use .iso?

**Comment by Raheemshah2809** [comment]
> > Just install python and do the 4 steps it's not that hard bro
> 
> It absolutely is, you just don't understand how un-tech-savy the average person is. Experience Hell

you're being lazy its literally one line of code to make an exe, use stack overflow or use chatgpt to help you.

**Comment by polartron** [comment]
> > > Just install python and do the 4 steps it's not that hard bro
> > 
> > 
> > It absolutely is, you just don't understand how un-tech-savy the average person is. Experience Hell
> 
> you're being lazy its literally one line of code to make an exe, use stack overflow or use chatgpt to help you.

If it's one line of code then why is it so hard to provide an exe. Checkmate atheist.

**Comment by Raheemshah2809** [comment]
> > > > Just install python and do the 4 steps it's not that hard bro
> > > 
> > > 
> > > It absolutely is, you just don't understand how un-tech-savy the average person is. Experience Hell
> > 
> > 
> > you're being lazy its literally one line of code to make an exe, use stack overflow or use chatgpt to help you.
> 
> If it's one line of code then why is it so hard to provide an exe? Checkmate, atheist.

Because everyone knows the people crying for an exe are people who are lazy and want the work done for them. learn to code sweetheart don't depend on others.

**Comment by idanarye** [comment]
> At this point just rewrite the whole shebang in c++ and watch the normies mald to their demise.
> 
> we will be there.

Rewrite it in node.js and have it depend on leftpad.

**Comment by FailureSam** [comment]
I want to be part of history...

**Comment by marsidev** [comment]
delete this

**Comment by yumio7** [comment]
![image](https://github.com/sherlock-project/sherlock/assets/67013996/2ac201e0-4f38-425d-bd5d-813323811a8c)

**Comment by kidofcubes** [comment]
Ought to make hardware for this so its plug and play :D

**Comment by FairyTail2000** [comment]
@NicusorN5 https://github.com/sherlock-project/sherlock/issues/2006#issuecomment-1952757454

> I originally made this issue as a tongue in cheek joke and nod to the Reddit post, but I'm gonna try to argue for it nevertheless.

**Comment by EnderNon** [comment]
> Best I can do is a `.bat` file:
> 
> ```batchfile
> git clone https://github.com/sherlock-project/sherlock.git
> cd sherlock
> pip install -r requirements.txt
> cd sherlock
> echo Please enter the username you want to search for:
> set /p userName=
> python sherlock.py %userName%
> echo Press any key to exit...
> pause
> ```
> 
> ~it's literally just the installation instructions~

might be possible to also install python. does windows have a native way of downloading files from a batch script?

**Comment by Lopoi** [comment]
> might be possible to also install python. does windows have a native way of downloading files from a batch script?

You can use `msiexec` to run the installers for python and git, but ideally you would want to check if they are installed first so you don't break stuff.

Also, you might need to refresh the cmd after installing them so it can recognize the new commands

**Comment by CallumCarmicheal** [comment]
> At this point just rewrite the whole shebang in c++ and watch the normies mald to their demise.
> 
> we will be there.


> What if we make it in C++ and force users to install 10 GB of build tools just to compile this project into a .exe file

You are all thinking too positive, we must make the build environment hell. It's not just about downloading the tools, we gotta make it hell by having each cpp file as their own library that cant be included. Put each function into its own C++ file, for even more fun make hell dependency tree then add a obvious recursive dependency at the end, something that most `nerds` could spot when compiling or reading the first line of the gcc error output.

```sh
g++ -c -o sher_help.o sher_help.cpp
g++ -c -o sher_parse_user_input.o sher_parse_user_input.cpp
g++ -c -o sher_process_arguments.o sher_process_arguments.cpp
g++ -c -o sher_process_domain.o sher_process_domain.cpp -I$CURL_INCLUDE_DIR -L$CURL_LIB_DIR -lcurl -Wl,-rpath=$CURL_LIB_DIR
ar rcs libsher_functions.a sher_help.o sher_parse_user_input.o sher_process_arguments.o sher_process_domain.o
g++ -o sherlock sher_main.cpp -L. -lsher_functions -Wl,-Bstatic -lsher_functions -Wl,-Bdynamic -L$CURL_LIB_DIR -lcurl -Wl,-rpath=$CURL_LIB_DIR
strip sherlock

# Compile each function separately into object files
# Compile sher_process_domain.cpp with Curl library and library path
# Create a static library from the object files
# Compile the main program and link it with both shared and static libraries
# `strip` to remove unnecessary symbols from the executable
```

**Comment by DontEatOreo** [comment]
> > Best I can do is a `.bat` file:
> > ```batchfile
> > git clone https://github.com/sherlock-project/sherlock.git
> > cd sherlock
> > pip install -r requirements.txt
> > cd sherlock
> > echo Please enter the username you want to search for:
> > set /p userName=
> > python sherlock.py %userName%
> > echo Press any key to exit...
> > pause
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > ~it's literally just the installation instructions~
> 
> might be possible to also install python. does windows have a native way of downloading files from a batch script?

You could just use a PowerShell script to check if `git` and `python` are installed and if not invoke `winget` to install them, or just force install `git` and `python` with `.bat`

**Comment by polartron** [comment]
> If you can't comprehend 3 fucking console commands, why are you even interested in using a CLI app?
> 
> Windows users these days...

When I order a sleezy greasy lean and mean pizza I don't want to be told to make it myself, kapish?

**Comment by SuperchupuDev** [comment]
i think the best solution here clearly would be a electron app that downloads the required exe. just think about the advantages of chromium: it could even allow to have a button to download it again!

**Comment by DontEatOreo** [comment]
> > If you can't comprehend 3 fucking console commands, why are you even interested in using a CLI app?
> > Windows users these days...
> 
> When I order a sleezy greasy lean and mean pizza I don't want to be told to make it myself, kapish?

If someone doesn't know how to navigate the terminal (or command prompt in the case of Windows), then they are not the target audience for Sherlock and **MOST** likely don't have the ethics of properly using this tool

**Comment by TailsFanLOL** [comment]
> > If you can't comprehend 3 fucking console commands, why are you even interested in using a CLI app?
> > Windows users these days...
> 
> When I order a sleazy greasy lean and mean pizza I don't want to be told to make it myself. Kapish?

Tf you mean

**Comment by Aurillium** [comment]
> i think the best solution here clearly would be a electron app that downloads the required exe. just think about the advantages of chromium: it could even allow to have a button to download it again!

I see your point, but Electron apps are resource intensive. I think instead we should rewrite it in pure assembly as its own operating system, for every CPU possible. This has the added advantage of controlling all dependencies because we implement them ourselves. And if you don’t want to dual-boot to use it, you can virtualise!

**Comment by idanarye** [comment]
> i think the best solution here clearly would be a electron app that downloads the required exe. just think about the advantages of chromium: it could even allow to have a button to download it again!

Instead of Chromium, why not use proper Chrome? Get spied on while you spy on others.

**Comment by akpi816218** [comment]
Hey! How come no one's talking about packaging a .DEB? While you're at it, how about an AppImage, Flatpak, RPM, zip, tarball run thru xzip, tarball run thru GZIP,  some brain DMG, some emotional DMG, and just a plain old binary that works only on 64-bit linux-based systems?

**Comment by ShobanChiddarth** [comment]
> Let them cook.
> 
> https://github.com/Chizaruu/sherlock/releases/tag/v69

The .exe does nothing. Just tried running it.

**Comment by idanarye** [comment]
> > Let them cook.
> > https://github.com/Chizaruu/sherlock/releases/tag/v69
> 
> The .exe does nothing. Just tried running it.

The .exe does nothing _that you know of_.

**Comment by akpi816218** [comment]
> > Let them cook.
> > 
> > https://github.com/Chizaruu/sherlock/releases/tag/v69
> 
> The .exe does nothing. Just tried running it.

might have been the whole point.

**Comment by WeaveAche** [comment]
Where's the build for my RISC-V machine running TempleOS??

**Comment by ShobanChiddarth** [comment]
@idanarye 
> > > Let them cook.
> > > https://github.com/Chizaruu/sherlock/releases/tag/v69
> > 
> > 
> > The .exe does nothing. Just tried running it.
> 
> The .exe does nothing _that you know of_.

[sherlock-does-not-work.webm](https://github.com/sherlock-project/sherlock/assets/68191453/a027fe4a-58e7-4c85-80b7-a38cabcc1768)

Alright so what does it do? It doesn't even print anything. Other .exe files work just fine. I installed librewolf with .exe, installed Bonzi Buddy with that .exe (I know it is a spyware, I just wanted to try how it looked like out of curiosity, since it was mentioned here), installed virtualbox guest additions with that .exe. They all worked just fine.

**Comment by ShobanChiddarth** [comment]
@akpi816218 
> > > Let them cook.
> > > https://github.com/Chizaruu/sherlock/releases/tag/v69
> > 
> > 
> > The .exe does nothing. Just tried running it.
> 
> might have been the whole point.
400 mb for nothing? I expected at least a rick roll.

**Comment by RigglePrime** [comment]
> Let them cook.
> 
> https://github.com/Chizaruu/sherlock/releases/tag/v69

> 400MB

Someone forgot to build in a virtual environment

**Comment by Raheemshah2809** [comment]
> > If you can't comprehend 3 fucking console commands, why are you even interested in using a CLI app?
> > Windows users these days...
> 
> When I order a sleazy greasy lean and mean pizza I don't want to be told to make it myself. Kapish?

You paid for that so you're expecting a good or service, this is given to you for free, make it yourself boo xoxo

**Comment by SuperchupuDev** [comment]
> > If you can't comprehend 3 fucking console commands, why are you even interested in using a CLI app?
> > 
> > Windows users these days...
> 
> When I order a sleazy greasy lean and mean pizza I don't want to be told to make it myself. Kapish?

that can happen when you order pizza as a normal user, have you tried sudo make me a pizza?

**Comment by suobset** [comment]
> Why stop at packaging it as an exe, bat, or msi? We should support .scr too!

Let's just rewrite it in HolyC for [TempleOS](https://templeos.org/) while we're at it

**Comment by Wengiel31** [comment]
People here talking about rewriting this in Rust... instead, we should focus on turning this into a website with prue JS without any frameworks and have it be pre ES6 of course. Newer versions are too bloated

**Comment by AaronTechnic** [comment]
It's strictly a CLI program, I doubt you could still use it even with an exe.

**Comment by idanarye** [comment]
> > It's strictly a CLI program, I doubt you could still use it even with an exe.
> 
> EXE CLI programs are totally a thing tho, if you double click one it just opens a terminal

You'd have to change it though so that it'll get the user name from STDIN instead of a CLI argument, and that it won't close automatically after finishing. Also, I doubt script kiddies who need EXEs use terminals with clickable links...

**Comment by EnderNon** [comment]
> > might be possible to also install python. does windows have a native way of downloading files from a batch script?
> 
> You can use `msiexec` to run the installers for python and git, but ideally you would want to check if they are installed first so you don't break stuff.
> 
> Also, you might need to refresh the cmd after installing them so it can recognize the new commands

maybe this would work

`@echo off
REM so basically this code starts at main. 
REM go to :checkgit. check if git is installed. If it is go to :checkpy, otherwise, go to :installgit and then go to :checkpy. You end up in :checkpy anyway.
REM go to :checkpy, check if python is installed. if it is, go straight to :main2, otherwise, go to :installgit first and then go to :main2. you end up in :main2 later, which is just code that goes after the install git and py functions

:main
echo SOME CODE HERE IN main
goto :checkgit

:checkgit
git --version >nul 2>&1 && ( echo "found Git" && goto :checkpy) || ( echo "Could not find git, installing" && call :installgit)

:checkpy
python --version >nul 2>&1 && ( echo "found Python" && goto :main2 ) || ( echo "Could not find Python, installing" && call :installpy)

:installgit
echo CODE TO INSTALL GIT GOES HERE
goto :checkpy
:installpy
echo CODE TO INSTALL PYTHON GOES HERE
goto :main2


:main2
echo SOME MORE CODE HERE IN main2`

**Comment by EnderNon** [comment]
actually hang on if i'm going to try and install git and python i'd need to check if I have admin

**Comment by sprucecellodev125** [comment]
> Let them cook.
> 
> https://github.com/Chizaruu/sherlock/releases/tag/v69

nice release
I'm subbing this issue

anyway

> > Just install python and do the 4 steps it's not that hard bro
> 
> It absolutely is, you just don't understand how un-tech-savy the average person is. Experience Hell


if you have Chocolatey, you can use 5 commands to install Git, Python and this app (and run it) in Windows, or more better, you can use Linux. Just

```bash
choco install python git
git clone https://github.com/sherlock-project/sherlock.git
cd sherlock
pip install -r requirements.txt
py sherlock.py userName
```

or

```bash
sudo apt install python3 python3-pip git
git clone https://github.com/sherlock-project/sherlock.git
cd sherlock
pip install -r requirements.txt
python3 sherlock.py userName
```

It isn't hard right? (Note: you need Admin access to do this (it isn't hard too) just search CMD, right click at first results, and "Run as Administrator")

**Comment by Lopoi** [comment]
I can't belive I spent my night on this, but here is your damn [`.exe`](https://github.com/Lopoi/watson/releases/tag/beta)

https://github.com/Lopoi/watson

**Comment by jthoward64** [comment]
> I can't belive I spent my night on this, but here is your damn [`.exe`](https://github.com/Lopoi/watson/releases/tag/beta)
> 
> https://github.com/Lopoi/watson

I DONT GIVE A FUCK ABOUT THE FUCKING ANITVIRUS! i just want to download this stupid fucking application and use it WHY IS THERE A ZIP??? MAKE A FUCKING .EXE FILE AND GIVE IT TO ME. these dumbfucks think that everyone is a developer and understands security. well i am not and i don't understand it. I only know to download and install applications. SO WHY THE FUCK IS THERE A ZIP? make an EXE file and give it to me. STUPID FUCKING SMELLY NERDS

*/s*

**Comment by Chizaruu** [comment]
> I can't believe I spent my night on this, but here is your damn [`.exe`](https://github.com/Lopoi/watson/releases/tag/beta)
> 
> https://github.com/Lopoi/watson

> The Window will probably freeze while it's searching, it is a feature.

Lmao, yeah I noticed that when I was testing it, the program literally eats up as much of the CPU as possible.

**Comment by Chizaruu** [comment]
> @idanarye 
> > > > Let them cook.
> > > > https://github.com/Chizaruu/sherlock/releases/tag/v69
> > > 
> > > 
> > > The .exe does nothing. Just tried running it.
> > 
> > The .exe does nothing _that you know of_.
> 
> [sherlock-does-not-work.webm](https://github.com/sherlock-project/sherlock/assets/68191453/a027fe4a-58e7-4c85-80b7-a38cabcc1768)
> 
> Alright so what does it do? It doesn't even print anything. Other .exe files work just fine. I installed librewolf with .exe, installed Bonzi Buddy with that .exe (I know it is spyware, I just wanted to try how it looked out of curiosity since it was mentioned here), and installed VirtualBox guest additions with that .exe. They all worked just fine.

Idk works on my machine. 

(Windows 11 Pro)

The exe isn't made for peeps with extensive knowledge that can fully understand the sophisticated intricacies of establishing a safe, and secure virtual sandbox that would limit both the tracking capabilities and potential malicious activity that this program may give way to, note that your sandbox environment may be hindering the processes of part or all of the packaged code/libraries within the exe

TL;DR; virtual environment issues suck 🤷‍♂️

**Comment by DanSM-5** [comment]
Since the whole exe gave an idea, I've forked sherlock to add a setup.py script so now it can be installed with pipx:

```bash
pipx install git+https://github.com/DanSM-5/sherlock.git@master
```

Best part is that pipx will create a exe for it:

```powershell
where.exe sherlock
# ~\.local\bin\sherlock.exe
```

**Comment by ShobanChiddarth** [comment]
@Chizaruu 
> > @idanarye
> > > > > Let them cook.
> > > > > https://github.com/Chizaruu/sherlock/releases/tag/v69
> > > > 
> > > > 
> > > > The .exe does nothing. Just tried running it.
> > > 
> > > 
> > > The .exe does nothing _that you know of_.
> > 
> > 
> > 
> >   
> >     
> >     
> > 
> >     sherlock-does-not-work.webm
> >     
> >   
> > 
> >   
> > 
> >   
> > 
> > 
> > Alright so what does it do? It doesn't even print anything. Other .exe files work just fine. I installed librewolf with .exe, installed Bonzi Buddy with that .exe (I know it is spyware, I just wanted to try how it looked out of curiosity since it was mentioned here), and installed VirtualBox guest additions with that .exe. They all worked just fine.
> 
> Idk works on my machine.
> 
> (Windows 11 Pro)
> 
> The exe isn't made for peeps with extensive knowledge that can fully understand the sophisticated intricacies of establishing a safe, and secure virtual sandbox that would limit both the tracking capabilities and potential malicious activity that this program may give way to, note that your sandbox environment may be hindering the processes of part or all of the packaged code/libraries within the exe
> 
> TL;DR; virtual environment issues suck 🤷‍♂️

Works fine in a Debian VM (with more RAM than that Windows VM) via wine. I didn't run it in bare metal Windows because of the sentence "Imagine trusting random code built by someone on the internet. :/" in the release page.

**Comment by Chizaruu** [comment]
@ShobanChiddarth 
> Works fine in a Debian VM (with more RAM than that Windows VM) via wine. I didn't run it in bare metal Windows because of the sentence "Imagine trusting random code built by someone on the internet. :/" in the release page.

The sentence is to highlight an issue that users have where they just use technology without making sure it's safe.

To my knowledge, Sherlock is "safe".

**Comment by ShobanChiddarth** [comment]
> @ShobanChiddarth
> 
> > Works fine in a Debian VM (with more RAM than that Windows VM) via wine. I didn't run it in bare metal Windows because of the sentence "Imagine trusting random code built by someone on the internet. :/" in the release page.
> 
> The sentence is to highlight an issue that users have where they just use technology without making sure it's safe.
> 
> To my knowledge, Sherlock is "safe".

K but I suspected you put something in it while creating that exe, like rick roll or any other jumpscare as a sort of prank, or worse yet delete the System32 folder or something that will harm the system.

**Comment by Chizaruu** [comment]
> K but I suspected you put something in it while creating that exe, like rick roll or any other jumpscare as a sort of prank, or worse yet delete the System32 folder or something else that will harm the system.

I was tempted to rickroll, to be honest. However, it was an excellent opportunity to properly introduce the Twitter/Reddit audience that was curious about what Sherlock is to the software.

**Comment by ilia-luk** [comment]
Future kids of mine, know papi was here when this part of history took place.

**Comment by Blu3u** [comment]
I was there Gandalf, I was there 3000 commits ago...

**Comment by ejsadiarin** [comment]
classic i was here moment

**Comment by CallumCarmicheal** [comment]
> @idanarye
> 
> > > > Let them cook.
> > > > https://github.com/Chizaruu/sherlock/releases/tag/v69
> > > 
> > > 
> > > The .exe does nothing. Just tried running it.
> > 
> > 
> > The .exe does nothing _that you know of_.
> 
>  sherlock-does-not-work.webm 
> Alright so what does it do? It doesn't even print anything. Other .exe files work just fine. I installed librewolf with .exe, installed Bonzi Buddy with that .exe (I know it is a spyware, I just wanted to try how it looked like out of curiosity, since it was mentioned here), installed virtualbox guest additions with that .exe. They all worked just fine.

No one noticed it, so I'll let you know what happened. In a Windows environment, whenever you highlight or select text in a command prompt, it blocks the next write operation, waiting for an interrupt until the user has selected and copied the text. The white cursor you see on the window puts the CMD window into select mode, locking the thread that next writes to stdout until the user has finished their selection.

All that to say, it works; you just accidentally blocked the app from continuing. Reality is often more disappointing than a delayed Rickroll.

**Comment by JulianH99** [comment]
To be honest, I think this was really disrespectful
@zimonitrome is open source work a joke to you? a mean to make funny reddit posts? do you think it's ok to make a post like this and take other peoples time just to entertain yourself

**Comment by idanarye** [comment]
> do you think it's ok to make a post like this and take other peoples time just to entertain yourself

To be fair - this was entertaining many people, not just OP.

**Comment by suobset** [comment]
Let’s not take life too seriously at times, @JulianH99. While this thread and any subsequent issues should def be locked (I believe just this one thread was enough and do not endorse the new threads after this one), Open Source is full of hidden gems and Easter eggs. That Reddit post was entertaining solely due to the fact that we have all had at least one such entitled person disrespect the community. In a way, this is us laughing at these entitled people. 

I do think the subsequent issues are uncalled for, this joke should’ve ended with this one issue. 

Life’s beautiful. Time to lock this stupid issue, and any subsequent stupid issues that reference to that Reddit post.

**Comment by Mavox** [comment]
> > I can't belive I spent my night on this, but here is your damn [`.exe`](https://github.com/Lopoi/watson/releases/tag/beta)
> > https://github.com/Lopoi/watson
> 
> I DONT GIVE A FUCK ABOUT THE FUCKING ANITVIRUS! i just want to download this stupid fucking application and use it WHY IS THERE A ZIP??? MAKE A FUCKING .EXE FILE AND GIVE IT TO ME. these dumbfucks think that everyone is a developer and understands security. well i am not and i don't understand it. I only know to download and install applications. SO WHY THE FUCK IS THERE A ZIP? make an EXE file and give it to me. STUPID FUCKING SMELLY NERDS
> 
> _/s_

And why should anyone cater to your needs, especially with your childish behaviour? Learn the required technologies or GTFO.

**Comment by AaronTechnic** [comment]
> > It's strictly a CLI program, I doubt you could still use it even with an exe.
> 
> EXE CLI programs are totally a thing tho, if you double click one it just opens a terminal

Ah, I didn't mean it like that, I meant that even as an EXE I doubt that the OP of the issue thread knows how to navigate through a CLI interface.

**Comment by johnvittu** [comment]
Reddit brought me here. I just wanted to say, that this issue is very much visible on other Github projects as well. Even though Python is so easy to understand, learn and use, people are still struggling and it's hilarious. 

But especially for a project like this (basically a doxxing script) I also agree that there should be at least some kind of entry level for the use of this. Even though it's obviously not a good thing to use same nickname on every service and website, but people have still done it and sometimes they can't even change their usernames anymore such as here on Github.

**Comment by SpottyTheTurtle** [comment]
Woe be the day that will come too soon when anyone can link chat gpt a github repo and have it do all the heavy lifting for them.

**Comment by faustinoaq** [comment]
Saving my place here for this hilarious moment in my developer life 🤣

**Comment by johnvittu** [comment]
> Woe be the day that will come too soon when anyone can link chat gpt a github repo and have it do all the heavy lifting for them.

Oh, the harsh heavy lifting of copying a command and pasting it into the console. How on earth can people demand something like this from a user, this is just unacceptable!

**Comment by winnie334** [comment]
What's the big problem with providing an exe? Just rename sherlock.py to sherlock.exe

**Comment by ALEZ-DEV** [comment]
Guys, I made a [GUI version](https://github.com/ALEZ-DEV/Sherlock-GUI) of Sherlock, give it a look ! 
It's super simple to use it !
It even come with a pre-compiled EXE file, that's not wonderful ?

**Comment by DanSM-5** [comment]
> Guys, I made a [GUI version](https://github.com/ALEZ-DEV/Sherlock-GUI) of Sherlock, give it a look ! It's super simple to use it ! It even come with a pre-compiled EXE file, that's not wonderful ?

Nice try but that's a Rick roll
https://github.com/ALEZ-DEV/Sherlock-GUI/blob/24d408035a6926bec09597d82a20f2b5790d3bb8/src/main.rs#L4

Don't get fooled guys

**Comment by jthoward64** [comment]
Alright, I think it's definitely time to lock and close this and #2019, the joke was fun but let's not get bogged down with it

**Comment by IOLOII** [comment]
https://github.com/sherlock-project/sherlock/issues/2011

**Comment by sprucecellodev125** [comment]
we got 62 participant here nice

**Comment by Maxz963** [comment]
63 participants now. Also, the CLI tool is actually really helpful for finding old accounts I've made so I can delete them.

**Comment by akpi816218** [comment]
> 63 participants now. Also, the CLI tool is actually really helpful for finding old accounts I've made so I can delete them. 

I haven't actually used this, but now you've convinced me to.

**Comment by RongDuJiKsp** [comment]
> > 此时，只需用 C++ 重写整个 shebang，然后看着正常人走向灭亡。
> > 我们会在那儿。
> 
> 让我们用 Rust 重写它 🦀🦀🦀

This is really a good idea 🦀🦀🦀🦀🦀🦀🦀🦀🦀

**Comment by RongDuJiKsp** [comment]
> > At this point just rewrite the whole shebang in c++ and watch the normies mald to their demise.
> > we will be there.
> 
> Let's rewrite it in rust 🦀🦀🦀

This is really a good idea 🦀🦀🦀🦀🦀🦀🦀🦀🦀

**Comment by whmsft** [comment]
why not use `nuitka` to build it?

**Comment by CallumCarmicheal** [comment]
> Guys, I made a [GUI version](https://github.com/ALEZ-DEV/Sherlock-GUI) of Sherlock, give it a look ! It's super simple to use it ! It even come with a pre-compiled EXE file, that's not wonderful ?

LGTM, Ship it.

**Comment by matheusfelipeog** [comment]
Hi everyone,

This is a bit overdue, but I didn't want to close without an official response.

We've considered creating a simple UI for the Sherlock project to make it easier for unfamiliar users, and there are some old issues and comments about it out there. Maybe we can do that in the future, but it's definitely not a priority right now since it's not our target audience.

Anyway, thank you very much for the serious considerations and the various jokes. It was really funny and enjoyable to follow; I had quite a few laughs.

Also, we received over 250K visits during the meme hype, which was definitely good for the project's marketing and outreach. Welcome to all new Sherlock users ;)

Thank you all, and until the next open source world meme. See you around.
